scripts/validate_docs_snippets.py

```diff
@@ -11,34 +11,44 @@
 from typing import List
 
 
-def run_docusaurus_build(tmp_dir_name: str) -> None:
-    # https://docusaurus.io/docs/cli#docusaurus-build-sitedir
+def check_dependencies(*deps: str) -> None:
+    for dep in deps:
+        if not shutil.which(dep):
+            raise Exception(f"Must have `{dep}` installed in PATH to run {__file__}")
+
+
+def run_docusaurus_build(target_dir: str) -> None:
     subprocess.call(
         [
             "yarn",
             "build",
             "--out-dir",
-            tmp_dir_name,
+            target_dir,
         ],
     )
 
 
 def run_grep(target_dir: str) -> List[str]:
-    out = subprocess.check_output(
-        [
-            "grep",
-            "-Enr",
-            r"<\/?snippet>",
-            target_dir,
-        ],
-        universal_newlines=True,
-    )
-    return out.splitlines()
+    try:
+        res = subprocess.run(
+            [
+                "grep",
+                "-Enr",
+                r"<\/?snippet>",
+                target_dir,
+            ],
+            capture_output=True,
+            universal_newlines=True,
+        )
+    except subprocess.CalledProcessError as e:
+        raise RuntimeError(
+            f"Command {e.cmd} returned with error (code {e.returncode}): {e.output}"
```

Comment:
Love the helpful debug output before re-raising. 💪  I would check with @billdirks but you might want to do a `from e`

Reply:
To also display the original exception in the traceback (Some info: https://stackoverflow.com/a/24752607/2916375)